drivers/sensors/qencoder.c

```diff
@@ -409,7 +409,7 @@ int qe_register(FAR const char *devpath, FAR struct qe_lowerhalf_s *lower)
    * (it was already zeroed by kmm_zalloc())
    */
 
-  nxsem_init(&upper->exclsem, 0, 1);
+  nxmutex_init(&upper->lock);
```

Comment:
I prefer that:

1. Keep ignore the error may be returned by register_driver if the origin code doesn't check it
2. Call nxmutex_destroy if the origin code free memory in case register_driver return fail